### ISSUE TITLE
Input changedetection not triggered when input is used as autocomplete-trigger and OnPush ChangeDetectionStrategy used 

### ISSUE THREAD

**Issue by diginikkari** [angular/material2#5271]
#### Bug, feature request, or proposal:
When using OnPush changedetection strategy, input with mdAutocomplete directive is not triggering changedetection when value is set in initialization.

#### What is the expected behavior?
When formcontrol value is set, changedetection should be triggered and placeholder text should be elevated.

#### What is the current behavior?
When formcontrol value is set, changedetection is not triggered. If there is placeholder text, it is not elevated when input value is set on initialization.

#### What are the steps to reproduce?
In linked plunker there is two md-inputs other with mdAutocomplete and other normal. Input placeholder text is elevated on normal md-input but not in input with autocomplete.

http://plnkr.co/edit/eLpDqe?p=preview

#### What is the use-case or motivation for changing an existing behavior?
To be able to use OnPush-changeDetectionstrategy.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 4.2.3
Material 2.0.0-beta.7

#### Is there anything else we should know?
nope

**Comment by willshowell** [comment]
Duplicate of https://github.com/angular/material2/issues/3005.

Also there is an upcoming effort to make all components OnPush https://github.com/angular/material2/issues/5035

**Comment by diginikkari** [comment]
Most of the inputs are already working in OnPush, but input used as MdAutocomplete-trigger is not yet working. Therefore I think this issue is valid.

**Comment by willshowell** [comment]
All the autocomplete-trigger is doing is taking the value you provide and setting the native input's value with the results of `displayWith()`. This happens after a [`Promise.resolve()`](https://github.com/angular/material2/blob/master/src/lib/autocomplete/autocomplete-trigger.ts#L234).

You can see in this plunker that it's the exact same thing with the input when you set the value after initializing it: http://plnkr.co/edit/4olO7E?p=preview

Basically it's all the same underneath. I acknowledge there is some trickery happening with the trigger manually floating and resetting the placeholder, but those methods are only run during the opening and closing of the panel and are not related to simply setting the value of the control.

Again, I may be wrong but when #3005 is fixed, it seems this too will be. 😃

**Comment by josephperrott** [comment]
Closing as it looks like this has been resolved if you are still seeing the issue feel free to open up an issue for us to take a look.
